dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+
+                            redirectAfterLogin = UtilMethods.isSet(request.getParameter(REFERRER_PARAMETER_KEY))
+                                    ? request.getParameter(REFERRER_PARAMETER_KEY) :
+                                    // this is safe, just to make a redirection when the user get's logged.
+                                    originalRequest;
+
+                            Logger.debug(this.getClass(),
+                                    "Executing SAML Login Redirection with request: " + redirectAfterLogin);
+
+                            // if we don't have a redirect yet
+                            if (null != session) {
+
+                                session.setAttribute(WebKeys.REDIRECT_AFTER_LOGIN, redirectAfterLogin);
+                                session.setAttribute(ORIGINAL_REQUEST,             originalRequest);
+                            }
+
+                            try {
+                                // this will redirect the user to the IdP Login Page.
+                                this.samlAuthenticationService.authentication(request, response, identityProviderConfiguration);
+                            } catch (Exception exception) {
+
+                                Logger.error(this,  "An error occurred when redirecting to the IdP Login page: " +
+                                        exception.getMessage(), exception);
+                                Logger.debug(this, ()-> "An error occurred when redirecting to the IdP Login page. Setting 500 " +
+                                        "response status.");
+                                response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                            }
+
+                            return Result.SKIP_NO_CHAIN;
+                        }
+                    }
+
+                    // Starting the logout
+                    // if it is logout
+                    Logger.debug(this, ()-> "----------------------------- doFilter --------------------------------");
+                    Logger.debug(this, ()-> "- isLogoutNeed = " + isLogoutNeed);
+                    Logger.debug(this, ()-> "- httpServletRequest.getRequestURI() = " + request.getRequestURI());
+
+                    if (isLogoutNeed && session != null && this.isLogoutRequest(request.getRequestURI(),
+                            this.getLogoutPathArray(identityProviderConfiguration))) {
+
+                        if (this.doLogout(response, request, session, identityProviderConfiguration)) {
+
+                            return Result.SKIP_NO_CHAIN;
+                        }
+                    }
+
+                } else {
+
+                    Logger.info(this, "No idpConfig for site '" + request.getServerName()
+                            + "'. No SAML filtering for this request: " + request.getRequestURI());
+                }
+            }
+        } catch (final Exception exception) { // todo: better error handling?
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)